src/index.c

```diff
@@ -883,41 +882,39 @@ static int NI_SkipTo(void *ctx, t_docId docId, RSIndexResult **hit) {
   return INDEXREAD_OK;
 }
 
-typedef struct NI_CriteriaTesterCtx{
-  IndexCriteriaTester* child;
-}NI_CriteriaTesterCtx;
+typedef struct NI_CriteriaTester {
+  IndexCriteriaTester base;
+  IndexCriteriaTester *child;
+} NI_CriteriaTester;
 
-static int NI_TextCriteria(void *ctx, t_docId id){
-  NI_CriteriaTesterCtx* ctc = ctx;
-  return !ctc->child->TextCriteria(ctc->child->ctx, id);
+static int NI_Text(struct IndexCriteriaTester *ct, t_docId id) {
```

Comment:
If there's a typedef you can leave out the `struct` tag. If there isn't a typedef, make one.